src/magic_folder/web.py

```diff
@@ -357,6 +358,23 @@ def list_all_sanpshots(self, request):
         _application_json(request)
         return json.dumps(dict(_list_all_snapshots(self._global_config)))
 
+    @app.route("/magic-folder/<string:folder_name>/scan", methods=['PUT'])
+    @inline_callbacks
+    def scan_folder(self, request, folder_name):
+        try:
+            folder_service = self._global_service.get_folder_service(folder_name)
+        except KeyError:
+            returnValue(NoResource(b"{}"))
+
+        body = _load_json(request.content.read())
+        if body != {"wait-for-snapshots": True}:
```

Comment:
Seems weird to have an option if we only ever do one thing...

Reply:
My feeling is that we may not always want this behaviour, and if that is the case, there is a reasonable chance to the default may not be the current behaviour. This way, we can add an option in the future, and have the default be whatever we want, and old clients will still get the current behaviour (that is, we can backwards compatibly add this option with whatever default we want).

Reply:
There's no use-case for it. If we only have one behavior, it doesn't need an option. (We can always add any _new_ option we want in the future and give it an appropriate default, whether this or anything else).

Reply:
We *can't* pick whatever default we want, without changing the behavior. The only way to add a new option without causing a change of behavior, is for the default is for to be the existing behavior. We can also *remove* this option in the future, if we decide the current behavior is the only one we want to support, as long as we continue to accept (and ignore) the option.

Reply:
So you're saying you want to keep this option now, even though it's essentially hard-coded, because you believe we might in the future want to change the endpoint to by default not wait for snapshots...?

Reply:
Exactly.

Reply:
The use case I have in mind is when the caller wants fire-and-forget behavior, they want to trigger a scan, but don't want to wait for the results. That feels like it is probably the most common use case. I needed/wanted a endpoint that waited for snapshots in the tests, so that is what I implemented; it was also easier to reason about not having the possibility of queuing the same file to be snapshoted twice, before the first was complete

Reply:
Sure, waiting is fine. Why can't we just add an option _if_ we want that in future ("fire and forget") instead of cluttering it now?

Reply:
We definitely could. I'd slightly prefer that to be the default behavior though. If the clutter wasn't entirely localized to the function (and its callers), or we couldn't just drop the option, if we don't want to keep it, I wouldn't suggest the option. As it is, it feels like keeping the option of having sensible defaults in the future has minimal maintenance burden (both immediate and ongoing).

Reply:
I probably would have left this option out and added something later when we found we needed it.  Since it's already here though, I don't see a lot of value in removing it.

Reply:
The "clutter" here is in the public HTTP API not just internal .. so now GridSync has to hard-code use of this option too (for example).